python/glow/functions.py

```diff
@@ -490,10 +490,10 @@ def sample_call_summary_stats(genotypes: Union[Column, str], refAllele: Union[Co
 
     Examples:
         >>> sites = [
-        ... Row(refAllele='C', alternateAlleles=['G'], genotypes=[Row(sampleId='NA12878', calls=[0, 0])]),
-        ... Row(refAllele='A', alternateAlleles=['G'], genotypes=[Row(sampleId='NA12878', calls=[1, 1])]),
-        ... Row(refAllele='AG', alternateAlleles=['A'], genotypes=[Row(sampleId='NA12878', calls=[1, 0])])]
-        >>> df = spark.createDataFrame(sites, 'alternateAlleles: array<string>, genotypes: array<struct<sampleId: string, calls: array<int>>>, refAllele: string')
+        ... {'refAllele': 'C', 'alternateAlleles': ['G'], 'genotypes': [{'sampleId': 'NA12878', 'calls': [0, 0]}]},
```

Comment:
PySpark stopped sorting Row fields in Spark 3.x, so this broke the tests. See https://spark.apache.org/docs/3.0.1/pyspark-migration-guide.html: we would have to set the environment variable PYSPARK_ROW_FIELD_SORTING_ENABLED to true for both executors and driver.

Reply:
We need to document this for developers.

Reply:
We should point to the migration guides in the release notes, but I'm not sure where else to document this. What do you think?

Reply:
notebooks are not working with the new version of spark 3.1.2
Did you test the notebooks with this version of Spark?

When reading a VCF with Glow v1.1.0 and
`PYSPARK_ROW_FIELD_SORTING_ENABLED=true`

I get:

`java.lang.NoSuchMethodError: org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformAllExpressions(Lscala/PartialFunction;)Lorg/apache/spark/sql/catalyst/plans/logical/LogicalPlan;`

Any idea how to resolve this?

Reply:
Also the notebook tests only test changed notebooks, but for a new version would want to change all of them?

Reply:
this was caused by a regression in Spark, which will be resolved soon
https://databricks.atlassian.net/browse/SC-84055
will continue with the release once this is in